### ISSUE TITLE
[no-namespace] Adding 'export' yields false positive when allowDeclarations=true

### ISSUE THREAD

**Issue by zachkirsch** [typescript-eslint/typescript-eslint#4843]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
        "@typescript-eslint/no-namespace": [
            "error",
            {
                allowDeclarations: true,
            },
        ],
```

This doesn't produce any warnings if `allowDeclarations` is true

```ts
declare namespace foo {
    namespace bar {
        namespace baz {}
    }
}
```

This DOES produce a warning if `allowDeclarations` is true

```ts
declare namespace foo {
    export namespace bar {
        namespace baz {}
    }
}
```

**Expected Result**

If `allowDeclarations` is true, then neither should error.

**Actual Result**

Including `export` causes the rule to complain.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.17.0` |
| `@typescript-eslint/parser`        | `5.17.0` |
| `TypeScript`                       | `4.6.3` |
| `ESLint`                           | `8.12.0` |
| `node`                             | `17.5.0` |
